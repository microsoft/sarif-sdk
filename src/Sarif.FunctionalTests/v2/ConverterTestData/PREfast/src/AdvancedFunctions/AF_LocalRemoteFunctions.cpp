#include "AF_LocalRemoteFunctions.h"

void LocalFunction(int* x);

void LocalFunction(int* x, int* y);

int UseLocalFunction(int * p)
{
	if (p == 0)
		int x = 5;
	LocalFunction(p);
    RemoteFunction(p, p);
	return *p;
}
