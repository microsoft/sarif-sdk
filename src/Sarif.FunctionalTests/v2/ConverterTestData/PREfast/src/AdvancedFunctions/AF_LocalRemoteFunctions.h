
void RemoteFunction(int* x);

void RemoteFunction(int* x, int* y);
