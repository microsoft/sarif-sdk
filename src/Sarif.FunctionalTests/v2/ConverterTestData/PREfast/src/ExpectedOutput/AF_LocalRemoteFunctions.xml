<?xml version='1.0' encoding='UTF-8'?>
<DEFECTS><DEFECT><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedfunctions\</FILEPATH><FILENAME>af_localremotefunctions.cpp</FILENAME><LINE>13</LINE><COLUMN>1</COLUMN></SFA><DEFECTCODE>6011</DEFECTCODE><DESCRIPTION>Dereferencing NULL pointer 'p'. </DESCRIPTION><FUNCTION>UseLocalFunction</FUNCTION><DECORATED>?UseLocalFunction@@YAHPAH@Z</DECORATED><FUNCLINE>7</FUNCLINE><PROBABILITY>1</PROBABILITY><RANK>4</RANK><CATEGORY><RULECATEGORY>mspft</RULECATEGORY></CATEGORY><PATH><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedfunctions\</FILEPATH><FILENAME>af_localremotefunctions.cpp</FILENAME><LINE>9</LINE><COLUMN>7</COLUMN><KEYEVENT><ID>1</ID><KIND>branch</KIND><IMPORTANCE>Full</IMPORTANCE><MESSAGE>'p' may be NULL (Enter this branch)</MESSAGE></KEYEVENT></SFA><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedfunctions\</FILEPATH><FILENAME>af_localremotefunctions.cpp</FILENAME><LINE>10</LINE><COLUMN>6</COLUMN></SFA><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedfunctions\</FILEPATH><FILENAME>af_localremotefunctions.cpp</FILENAME><LINE>11</LINE><COLUMN>14</COLUMN><KEYEVENT><ID>2</ID><KIND>declaration</KIND><IMPORTANCE>Full</IMPORTANCE><MESSAGE>'p' is an In/Out parameter to 'LocalFunction' (declared on line 3)    (Stmt: LocalFunction(p))</MESSAGE></KEYEVENT></SFA><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedfunctions\</FILEPATH><FILENAME>af_localremotefunctions.cpp</FILENAME><LINE>12</LINE><COLUMN>18</COLUMN><KEYEVENT><ID>3</ID><KIND>declaration</KIND><IMPORTANCE>Full</IMPORTANCE><MESSAGE>'p' is an In/Out parameter to 'RemoteFunction' (declared at e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedfunctions\af_localremotefunctions.h:4)    (Stmt: RemoteFunction(p, p))</MESSAGE></KEYEVENT></SFA><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedfunctions\</FILEPATH><FILENAME>af_localremotefunctions.cpp</FILENAME><LINE>13</LINE><COLUMN>1</COLUMN><KEYEVENT><ID>4</ID><KIND>usage</KIND><IMPORTANCE>Essential</IMPORTANCE><MESSAGE>'p' is dereferenced, but may still be NULL</MESSAGE></KEYEVENT></SFA></PATH></DEFECT></DEFECTS>